src/python/pants/backend/python/python_artifact.py

```diff
@@ -1,64 +1,43 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+from typing import Any, Dict
+
 
 class PythonArtifact:
     """Represents a Python setup.py-based project."""
 
-    class MissingArgument(Exception):
-        pass
-
-    class UnsupportedArgument(Exception):
-        pass
-
-    UNSUPPORTED_ARGS = frozenset({"data_files", "package_dir", "package_data", "packages"})
-
     def __init__(self, **kwargs):
         """
         :param kwargs: Passed to `setuptools.setup
           <https://pythonhosted.org/setuptools/setuptools.html>`_.
         """
-        self._kw = kwargs
+        if "name" not in kwargs:
+            raise ValueError("PythonArtifact (aka `setup_py`) requires `name` to be specified.")
```

Comment:
I don't think we need to confuse the user with the internal name `PythonArtifact`. How about just setup_py() requires `name` to be specified`?